packages/legend-studio/src/stores/sidebar-state/WorkspaceWorkflowsState.ts

```diff
@@ -14,35 +14,208 @@
  * limitations under the License.
  */
 
-import { makeAutoObservable } from 'mobx';
+import type { TreeNodeData, TreeData } from '@finos/legend-art';
+import { makeAutoObservable, observable, action, flowResult } from 'mobx';
 import { STUDIO_LOG_EVENT } from '../StudioLogEvent';
 import type { EditorStore } from '../EditorStore';
 import type { EditorSdlcState } from '../EditorSdlcState';
 import type { GeneratorFn, PlainObject } from '@finos/legend-shared';
-import { assertErrorThrown, LogEvent } from '@finos/legend-shared';
-import { Workflow } from '@finos/legend-server-sdlc';
+import {
+  assertErrorThrown,
+  LogEvent,
+  isNonNullable,
+} from '@finos/legend-shared';
+import { WorkflowJob, Workflow } from '@finos/legend-server-sdlc';
+
+export abstract class WorkflowExplorerTreeNodeData implements TreeNodeData {
+  isSelected?: boolean | undefined;
+  isOpen?: boolean | undefined;
+  id: string;
+  label: string;
+  childrenIds: string[] | undefined;
+  constructor(id: string, label: string) {
```

Comment:
in some weird case, we might get clashing of IDs here, could we add some prefix to these IDs to differentiate them?

Reply:
you mean there might be cases where workflow id and workflow job ids are the same???

Reply:
seems a bit excessive but I guess there is not contract of uniqueness between those ids.

Reply:
yeah, I mean let's do `workflow_${id}` and `job_${id}`